### DISCUSSION TITLE
Import VS Code keyboard shortcuts

### DISCUSSION THREAD

**Discussion by davidpmccormick** [zed-industries/zed#6544] [Feature Requests]
Using a new editor highlights how dependent I've become on lots of shortcuts. In VS Code I toggle the dash open/closed with cmd-shift-i. I think the equivalent of this looks like shift-escape in zed, but after several minutes of trying to overwrite the default keybindings I couldn't get the same behaviour.

So I guess in the short term I'm looking for a way to get that specific shortcut to do what I'd like, but better yet I'd love a way to port all my shortcuts from VS Code more easily.

Thanks

**Comment by hovsater** [comment]
@davidpmccormick the following bindings would toggle the dock in Zed by pressing `⌘⇧I`:

```json
[
  {
    "context": "Workspace || Editor",
    "bindings": {
      "cmd-shift-i": "dock::FocusDock"
    }
  },
  {
    "context": "Pane && docked",
    "bindings": {
      "cmd-shift-i": "dock::HideDock"
    }
  }
]
```

With that said, the same binding is being used to format the current buffer, so you might want to remap that one after these changes. 🙂

**Comment by ashirbad29** [comment]
is there any way to port all my VS Code shortcuts to zed now ?

**Comment by venkatd** [comment]
I would love to have a utility for this! It's the reason why I have Zed installed but I keep procrastinating swapping over because wait I have to reconfigure all my key-bindings.

**Comment by ankit-aabad** [comment]
I wanted to try zed, but so used to my shortcuts that it is difficult. There should be a smooth migration from vscode to zed.

**Comment by videni** [comment]
This would be very helpful for users who come from the vscode world.

**Comment by mizrabsheikh** [comment]
An option to migrate keyboard shortcuts from VS Code would be Godsend 🙏
